src/include/sof/audio/component.h

```diff
@@ -262,6 +283,7 @@ struct comp_ops {
  */
 struct comp_driver {
 	uint32_t type;		/**< SOF_COMP_ for driver */
+	uint32_t uid;		/**< Address of uuid_entry */
```

Comment:
should it be *uid then? Or uintptr_t?

Reply:
Probably uintptr_t, the thing is we do not want to actually dereference this address because it points to a section that will not be uploaded to the DSP. Similar to trace messages.

Reply:
I was concerned with this type too. However, at some point the uid (named w/o double *u* to make a slight difference with *uuid* that is a part of the structure that *uid* points to) must be converted to a type with fixed size being a part of the interface consumed by a program compiled and running on another cpu. Since the *uid* is not interpreted internally by the dsp code, I decided to use the target type immediately and pass it through the trace stack. This probably could be an offset inside the section instead of absolute address, computed from a linker variable if possible, to justify the type used by interface and to avoid issues with 64-bit addressing however I am not sure if such flexibility is really required.

Reply:
Absolute address is fine (it is used for the trace format message and number of parameters), so I'd insist on uintptr_t. I'm not sure what the format is (I'll look real quick).

Reply:
Came back, it is stored as uint32_t in that header. I don't like that but that section will be in the 32-bit range anyway even if we potentially would move to 64-bit DSPs or other cores (Zephyr!), the address for these sections would be in memory that shouldn't be dereferenced.

Guess I'll allow uint32_t due to the precedent.